Formula/linkerd.rb

```diff
@@ -1,82 +1,43 @@
+    srcpath = buildpath/"src/github.com/linkerd/linkerd2"
+    srcpath.install buildpath.children - [buildpath/".brew_home"]
 
-    pkgshare.install buildpath/"config/linkerd.yaml" => "default.yaml"
+    cd srcpath do
+      system "bin/build-cli-bin"
+      bin.install "target/cli/darwin/linkerd"
 
-    etc.install "config" => "linkerd"
-    etc.install "disco" => "linkerd/disco"
-    libexec.install_symlink etc/"linkerd" => "config"
-    libexec.install_symlink etc/"linkerd/disco" => "disco"
+      # Install bash completion
+      output = Utils.popen_read("#{bin}/linkerd completion bash")
+      (bash_completion/"linkerd").write output
 
-    share.install "docs"
-  end
-
-  def post_install
-    (var/"log/linkerd").mkpath
-  end
+      # Install zsh completion
+      output = Utils.popen_read("#{bin}/linkerd completion zsh")
+      (zsh_completion/"linkerd").write output
 
-  plist_options :manual => "linkerd #{HOMEBREW_PREFIX}/etc/linkerd/linkerd.yaml"
-
-  def plist; <<~EOS
-    <?xml version="1.0" encoding="UTF-8"?>
-    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-    <plist version="1.0">
-    <dict>
-        <key>Label</key>
-        <string>#{plist_name}</string>
-        <key>WorkingDirectory</key>
-        <string>#{HOMEBREW_PREFIX}</string>
-        <key>ProgramArguments</key>
-        <array>
-            <string>#{opt_bin}/linkerd</string>
-            <string>#{etc}/linkerd/linkerd.yaml</string>
-        </array>
-        <key>RunAtLoad</key>
-        <true/>
-        <key>KeepAlive</key>
-        <true/>
-        <key>StandardErrorPath</key>
-        <string>#{var}/log/linkerd/linkerd.log</string>
-        <key>StandardOutPath</key>
-        <string>#{var}/log/linkerd/linkerd.log</string>
-    </dict>
-    </plist>
-  EOS
+      prefix.install_metafiles
+    end
   end
 
   test do
-    (testpath/"index.html").write "It works!"
+    run_output = shell_output("#{bin}/linkerd 2>&1")
+    assert_match "linkerd manages the Linkerd service mesh.", run_output
 
-    simple_http_pid = fork do
-      exec "python -m SimpleHTTPServer 9999"
-    end
-    linkerd_pid = fork do
-      exec "#{bin}/linkerd #{pkgshare}/default.yaml"
-    end
-
-    sleep 10
-
-    begin
-      assert_match "It works!", shell_output("curl -s -H 'Host: web' http://localhost:4140")
-      assert_match "Bad Gateway", shell_output("curl -s -I -H 'Host: foo' http://localhost:4140")
-    ensure
-      Process.kill("TERM", linkerd_pid)
-      Process.wait(linkerd_pid)
-      Process.kill("TERM", simple_http_pid)
-      Process.wait(simple_http_pid)
-    end
+    version_output = shell_output("#{bin}/linkerd version --client 2>&1")
+    assert_match "Client version: ", version_output
+    stable_resource = stable.instance_variable_get(:@resource)
+    assert_match stable_resource.instance_variable_get(:@specs)[:tag], version_output if build.stable?
```

Comment:
i think we also need to add a test to ensure `linkerd install` can run with error:
```diff
diff --git a/Formula/linkerd.rb b/Formula/linkerd.rb
index 85360b5203..34ffc5a2f9 100644
--- a/Formula/linkerd.rb
+++ b/Formula/linkerd.rb
@@ -39,5 +39,7 @@ class Linkerd < Formula
     assert_match "Client version: ", version_output
     stable_resource = stable.instance_variable_get(:@resource)
     assert_match stable_resource.instance_variable_get(:@specs)[:tag], version_output if build.stable?
+
+    system "#{bin}/linkerd", "install"
   end
 end
```

related: this brew formula builds a `linkerd` binary that cannot successfully run `linkerd install`

Reply:
Found the related issue, fix is up at linkerd/linkerd#2341. Please wait for that to merge, and for a subsequent `linkerd-2.2.1-stable` release.

Reply:
here's a more complete diff, changes include:
- fixes audit issue by changing description to not start with `Linkerd`
- upgrade to `stable-2.2.1` to pull in build fix (linkerd/linkerd2#2341)
- add `linkerd install` test to validate build completed correctly

```diff
diff --git a/Formula/linkerd.rb b/Formula/linkerd.rb
index 85360b5203..7060cc4881 100644
--- a/Formula/linkerd.rb
+++ b/Formula/linkerd.rb
@@ -1,10 +1,10 @@
 class Linkerd < Formula
-  desc "Linkerd configuration command-line utility"
+  desc "Command-line interface for the Linkerd service mesh"
   homepage "https://linkerd.io"
 
   url "https://github.com/linkerd/linkerd2.git",
-    :tag      => "stable-2.2.0",
-    :revision => "1cde1f083c5f245170f75e1974f3b20f7fa34d9c"
+    :tag      => "stable-2.2.1",
+    :revision => "5e47cb150a33150e5aeddc6672d8a64701a970de"
 
   depends_on "go" => :build
 
@@ -39,5 +39,7 @@ class Linkerd < Formula
     assert_match "Client version: ", version_output
     stable_resource = stable.instance_variable_get(:@resource)
     assert_match stable_resource.instance_variable_get(:@specs)[:tag], version_output if build.stable?
+
+    system "#{bin}/linkerd", "install"
   end
 end
```

Reply:
@siggy you've done all work up here. Not sure, I'm doing right thing by pushing changes.

Anyways, made those changes you suggested. Please let me know, if anymore changes need to done.

Reply:
@murarisumit definitely appreciate you shepherding this through the process!

I just tested the latest changes, everything looks good. Linkerd `stable-2.2.1` was just released, so I think this branch is good to go from my perspective.